components/omega/doc/design/OmegaV0ShallowWater.md

```diff
@@ -0,0 +1,660 @@
+
+The perpendicular vector component a vector field is defined as
+$$
+ \nabla^\perp {\bf F} \equiv
+ {\bf k} \times \nabla {\bf F}
+$$
+so that it points 90$^o$ to the left of ${\bf F}$ in a right-handed coordinate system. 
+
+In TRiSK, the native component for the prognostic velocity field is normal edge velocity $u_e$. The perpendicular component at the edge is computed diagnostically as
+([Ringler et al. 2010](https://www.sciencedirect.com/science/article/pii/S0021999109006780) eqn 24)
+
+$$
+u_e^\perp = \frac{1}{d_e} \sum_{e'\in ECP(e)} w_{e,e'} \, l_e \, u_{e'}.
+$$
+
+The perpendicular vector is computed from the edge-normal components of all the edges of the two neighboring cells of edge $e$, denoted as $e'\in ECP(e)$. For example, on a mesh with hexagons, there are a total of 10 edges $e'$ on the two cells neighboring edge $e$ (the original edge $e$ is not included).  The perpendicular velocity $u_e^\perp$ is used in the Coriolis force, which appears in the potential vorticity advection term.  The weighting coefficients $w_{e,e'}$ are carefully chosen to conserve potential vorticity, as described in
+[Thuburn et al. 2009](https://www.sciencedirect.com/science/article/pii/S0021999109004434)
+and
+[Ringler et al. 2010](https://www.sciencedirect.com/science/article/pii/S0021999109006780).
+
+#### 3.2.5. Perpendicular Gradient
+The gradient of a scalar at the middle of an edge, pointing tangentially along the edge (from one vertex to the other), is sometimes used. For example, the del2 formulation requires the perpendicular gradient of vorticity. This is called the perpendicular gradient because the standard gradient is normal to the edge.
+
+The perpendicular gradient maps a scalar at vertices to an edge-tangential vector component,
+$$
+\left(\nabla \omega_v\right)_e^\perp = \frac{1}{l_e} \sum_{v\in VE(e)} -t_{e,v}\omega_v.
+$$
+Like the standard gradient, in practice we drop the sign indicator $t_{e,v}$ and rewrite it using the index ordering,
+$$
+\left(\nabla \omega_v\right)_e^\perp = \frac{\omega_{v2} - \omega_{v1}}{l_e},
+$$
+where the positive vector ${\bf n}^\perp$ is 90$^o$ to the left of ${\bf n}$. The indices are ordered such that ${\bf n}^\perp$ points from $v_1$ to $v_2$, which corresponds to VerticesOnEdge(0,iEdge) and VerticesOnEdge(1,iEdge) in the code.
+
+
+#### 3.2.6. Cell to Edge Interpolation
+This is simply a mid-point averaging from cell centers to the adjoining edge.
+
+$$
+[h_i]_e = \frac{1}{2} \sum_{i\in CE(e)} h_i
+$$
+
+#### 3.2.7. Vertex to Edge Interpolation
+This is a mid-point averaging from vertices to the middle of the connecting edge.
+
+$$
+[\omega_v]_e = \frac{1}{2} \sum_{v\in VE(e)} \omega_v
+$$
+
+#### 3.2.8. Cell to Vertex Interpolation
+This is an area-weighted average at a vertex from the three surrounding cells. Here ${\tilde A}_v$ is the dual triangle area centered on the vertex, and ${\hat A}_{i,v}$ is the kite area of cell $i$ with the kite point at vertex $v$.
```

Comment:
`A_v` is defined without the tilde in the curl equation and is missing from the variables table.

Reply:
Done, thanks.